### ISSUE TITLE
[Work_Item] Add columns that describe provider entities in Billing Account and Sub Account

### ISSUE THREAD

**Issue by flanakin** [FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec#317]
## 1. **Problem Statement** *

> Describe the problem, issue, use case, or opportunity that this work item addresses.
> Include practitioner quotes illustrating real examples a) of questions being asked by practitioners and b) value unlocked by answering these questions, if available.
> - What is the problem?: Explain the context and why it needs resolution.
> - Impact: Describe how the problem affects users, systems, or the project.

Different providers have different "types" of entities that may be needed to represent the `BillingAccountId/Name` and `SubAccoundId/Name`. Some providers even have multiple values that may be used. For example, Microsoft has 3 different types of entities that could be used for `BillingAccountId` depending on the type of agreement.

## Use case(s)
- `Understand what type of entity the billing account or sub account is`
  - ❗ Need a description

## 2. **Objective** *
> State the objective of this work item. What outcome is expected?
> * Success Criteria: Define how success will be measured (e.g. metrics and KPIs).

Create provider-specific "type" columns that describe what `BillingAccountId` and `SubAccountId` are.

## 3. **Supporting Documentation** *

> Include links to supporting documents such as:
> - Data Examples: [Link to data or relevant files; DO NOT share proprietary information]
> - Related Use Cases or Discussion Documents: [Link to discussion]
> - PRs or Other References: [Link to relevant references]

See Microsoft BillingAccountType, SubAccountType ([documentation](https://aka.ms/costmgmt/exports/focus), [PR #287](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/287)).

Potential BillingAccountType values:
- "Management Account" (AWS)
- "Billing Account" (GCP)
- "Billing Account" (Microsoft EA)
- "Billing Profile" (Microsoft MCA/MPA)
- "Subscription" (Microsoft MOSA)

Potential SubAccountType values:
- "Member Account" (AWS)
- "Project" (GCP)
- "Subscription" (Microsoft)
- "Compartment" (OCI)

## 4. **Proposed Solution / Approach**

> Outline any proposed solutions, approaches, or potential paths forward. Do not submit detailed solutions; please keep suggestions high-level.
> 
> - Initial Ideas: Describe potential solution paths, tools, or technologies.
> - Considerations: Include any constraints, dependencies, or risks.
> - Feasibility: Include any information that helps quantify feasibility, such as perceived level of effort to augment the spec, or existing fields in current data generator exports.
> - Benchmarks: Are there established best practices for solving this problem available to practitioners today (e.g. mappings from existing CSP exports that are widely used)?

Following the `ResourceId` and `ReosurceType` pattern, add new `BillingAccountType` and `SubAccountType` columns.

## 5. **Epic or Theme Association**
> This section will be completed by the Maintainers.
>
> - Epic: [Epic Name]
> - Theme: [Theme Name, if applicable]

TBD

## 6. **Stakeholders** *

> List the main stakeholders for this issue.
> 
> - Primary Stakeholder: [Name/Role]
> - Other Involved Parties: [Names/Roles]

- Primary stakeholder: Michael Flanakin, Product Manager

<br>

### Give it a 👍 below ↴

Comments are welcome and encouraged!

**Comment by mike-finopsorg** [comment]
While this will lead to a lot of duplicated data, I do see some major benefits in having values that users are likely to be looking for and it builds the association to the Billing/Sub Account.

**Comment by marc-perreaut** [comment]
Maybe I don't understand well, but would the values of columns BillingAccountType  and SubAccountType be always the same for a given provider?
If yes, I find it not efficient from a data perspective.
Maybe there could be something outside the FOCUS dataset, like a manifest, to enable the documentation of the mapping between provider's data and FOCUS columns.

**Comment by flanakin** [comment]
@marc-perreaut Not necessarily. Specifically, BillingAccountType for Microsoft EA would be "Billing Account" and for MCA would be "Billing Profile". That's why I proposed adding this. It's trying to address the confusion where BillingAccountId does not refer to a "billing account".

With respect to SubAccountType, there's a question that pops up from time to time about whether that should be a subscription or a resource group in Azure. This would drive that clarity, but I'm 100% with you regarding it being the same value. I mainly included it for completeness and to force the conversation about how we want to handle it.

**Comment by jpradocueva** [comment]
Agreed to move it to release v1.1 during Members meeting on April 25.

**Comment by shawnalpay** [comment]
@flanakin 

Consider the following except from the [supporting content for Subaccount](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/blob/fbb9ddf9d9889a5fb794090c484d8b9ca4e55973/supporting_content/columns/subaccountid.md).

> FOCUS plans to include Level 1, Level 3 in the specification. Level 5 (resource name, resource id) are already in the specification.
>
> 
> | Provider | Level 1 | Level 2 | Level 3 | Level 4 | Level 5 | |
> |----------|---------|---------|---------|---------|---------|---------|
> | | Billing Account | Sub Account Group (optional) | Sub Account | | | |
> | AWS | Management Account | Org | Account | Resource Group (optional) | Resource | |
> | GCP | Billing Account | Folder | Project | | | |
> | MS EA | Billing Account (Invoice Scope) | Department | Enrollment Account | Subscription | Resource Group | Resource |
> | MS MCA | Billing Account | Billing Profile (Invoice Scope) | Invoice Section | Subscription | Resource Group | Resource |
> | MS PAYG | Billing Account | Subscription (Invoice Scope) | Resource Group | Resource | | |
> | MS Internal | Subscription (Invoice Scope) | Resource Group | Resource | | | |
> |OCI | Organization | Subscription (Invoice Scope) | Tenancy | Compartment | Resource |

According to this: 

 - Level 1 is already represented as `BillingAccountId`.
 - Level 2 is not in the spec.
 - Level 3 is already represented as `SubAccountId`.  
 - Level 4 is not in the spec.
 - Level 5 is not in the spec.
 - Level 6 (not explicitly called as such) is already represented as `ResourceId`.

Based on that, a few questions for you:

- Which of these levels represent the "Billing Account Type" and "Sub Account Type" columns you propose?  
- Is this depiction of the levels inaccurate, and if so, do we need to change them?
- I believe Level 4 and Level 5 would be handled if/when we implement #301; true?

**Comment by richwang99** [comment]
Resource Group in Azure is a critical information for cost allocation. Although it is not a billable item by itself, we use it together with Azure policy to enforce the tag of all resources within the resource group. 
Some resources may have the same name, but belong to different resource groups. So Resource Group name will be essential.

**Comment by mike-finopsorg** [comment]
APJ FOCUS User Group call 23/October attendees raised this as very important for their work.

**Comment by jpradocueva** [comment]
### Notes from the Maintainers' call on November 4:
**Context:** This task aims to enhance the billing structure by adding attributes that describe provider entities in billing accounts and sub-accounts, aiding practitioners in tracking and understanding multi-account setups.
**Level of Effort Required:** Low — The task primarily involves adding new columns to the existing model, which is a simple structural adjustment.
**Level of Impact:** Medium – Adding provider entity descriptions improves transparency and traceability, particularly in multi-account setups, enhancing the ability of practitioners to manage and report costs across complex account structures.

**Comment by jpradocueva** [comment]
### Summary from the Maintainers' call on Nov 25
Context:
This work item focuses on specifying how provider entities (e.g., billing accounts and sub-accounts) should be represented in the FOCUS dataset.
Maintainers Assigned:
To be confirmed.
Task Force Assigned:
Task Force 2 (TF2).

**Comment by ijurica** [comment]
@flanakin Just a note that Google also has the [concept of a Parent Billing Account and Subaccount](https://cloud.google.com/billing/docs/concepts#subaccounts). 
Tim or David could probably share a bit more insights on this when the time comes.

![image](https://github.com/user-attachments/assets/a03fde2a-a609-45ba-9b1f-ae14415e2175)

**Comment by jpradocueva** [comment]
### Action Items from TF-2 call on December 4:
- [ ] [[#317](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/issues/317)] Michael @flanakin : Update the PR based on the poll results by mid-January (e.g., January 7) to allow for discussion during the TF-2 meeting.
- [ ] [[#317](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/issues/317)] Michael @flanakin : Discuss the updated PR during the TF-2 meeting, and then present it and seek approval at the Members Meeting by January 23 (beginning of the Interim Consistency Review).

**Comment by jpradocueva** [comment]
### Action Items from the Maintainers' call on December 16:
- [ ] [[#317](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/issues/317)] Michael @flanakin : Update the proposal based on the results of the earlier poll and discussion.

**Comment by jpradocueva** [comment]
### Action Items from the Maintainers' call on Jan 6:
- [ ] [#317] TF2 Team - Reassess and discuss the updates in mid-January.

**Comment by shawnalpay** [comment]
Reopening this Work Item to handle for use case augmentation and clarification of normative requirements for these new columns.

**Comment by shawnalpay** [comment]
All Action Items completed!  Closing.
